src/modules/collection/sagas.ts

```diff
@@ -578,8 +619,50 @@ export function* collectionSaga(builder: BuilderAPI, catalyst: CatalystClient) {
 
       // 5. If any, open the modal in the DEPLOY step and wait for actions
 
+      const { itemsToDeploy, entitiesToDeploy }: { itemsToDeploy: Item[]; entitiesToDeploy: DeploymentPreparationData[] } = yield call(
+        getTPItemsAndEntitiesToDeploy,
+        collection,
+        tree,
+        content_hashes
+      )
+
+      // 5. If any, open the modal in the DEPLOY step and wait for actions
+      if (itemsToDeploy.length > 0) {
+        const modalMetadata: ApprovalFlowModalMetadata<ApprovalFlowModalView.DEPLOY> = {
+          view: ApprovalFlowModalView.DEPLOY,
+          collection,
+          items: itemsToDeploy,
+          entities: entitiesToDeploy
+        }
+        yield put(openModal('ApprovalFlowModal', modalMetadata))
+
+        // Wait for actions...
+        const {
+          failure,
+          cancel
+        }: { success: DeployEntitiesSuccessAction; failure: DeployEntitiesFailureAction; cancel: CloseModalAction } = yield race({
+          success: take(DEPLOY_ENTITIES_SUCCESS),
+          failure: take(DEPLOY_ENTITIES_FAILURE),
+          cancel: take(CLOSE_MODAL)
+        })
+
+        // If failure show error and exit flow
+        if (failure) {
+          throw new Error(failure.payload.error)
+
+          // If cancel exit flow
+        } else if (cancel) {
+          return
+        }
+      }
+
       // 6. If the collection was approved but it had a pending curation, approve the curation
 
+      const itemCurations: ItemCuration[] = yield select(getItemCurations, collection.id)
```

Comment:
Do we need all the item curations to change their status? I'm thinking on when we'll have pagination and multiple items to approve.